src/Utilities/results.jl

```diff
@@ -23,6 +23,123 @@ function get_fallback(model::MOI.ModelLike, ::MOI.ObjectiveValue)
+                              ci::MOI.ConstraintIndex{<:MOI.AbstractScalarFunction,
+                                                      <:MOI.Interval},
+                              T::Type,
+                              result_index::Integer)
+    constant = scalar_constant(T, MOI.get(model, MOI.ConstraintFunction(), ci))
+    set = MOI.get(model, MOI.ConstraintSet(), ci)
+    dual = MOI.get(model, MOI.ConstraintDual(result_index), ci)
+    if dual < zero(dual)
+        # The dual is negative so it is in the dual of the MOI.LessThan cone
+        # hence the upper bound of the Interval set is tight
+        constant -= set.upper
+    else
+        # the lower bound is tight
+        constant -= set.lower
+    end
+    return set_dot(constant, dual, set)
+end
+
+
+
+"""
+    dual_objective_value(model::MOI.ModelLike,
+                    F::Type{<:MOI.AbstractFunction},
+                    S::Type{<:MOI.AbstractSet},
+                    T::Type,
+                    result_index::Integer)
+
+Return the part of `DualObjectiveValue` due to `F`-in-`S` constraints using scalar
+type `T`.
+"""
+function dual_objective_value(model::MOI.ModelLike,
+                         F::Type{<:MOI.AbstractFunction},
+                         S::Type{<:MOI.AbstractSet},
+                         T::Type,
+                         result_index::Integer)
+    value = zero(T) # sum won't work if there are now constraints.
+    for ci in MOI.get(model, MOI.ListOfConstraintIndices{F, S}())
+        value += dual_objective_value(model, ci, T, result_index)
+    end
+    return value
+end
+
+function dual_objective_value(model::MOI.ModelLike,
+                              F::Type{MOI.VectorOfVariables},
+                              S::Type{<:MOI.AbstractVectorSet},
+                              T::Type,
+                              result_index::Integer)
+    # No constant in the function nor set so no contribution to the dual
+    # objective value.
+    return zero(T)
+end
+
+function is_ray(status::MOI.ResultStatusCode)
+    return status == MOI.INFEASIBILITY_CERTIFICATE ||
+           status == MOI.NEARLY_INFEASIBILITY_CERTIFICATE
+end
+
+"""
+    get_fallback(model::MOI.ModelLike, ::MOI.DualObjectiveValue, T::Type)::T
+
+Compute the dual objective value of type `T` using the `ConstraintDual` results
+and the `ConstraintFunction` and `ConstraintSet` values.
+"""
+function get_fallback(model::MOI.ModelLike, attr::MOI.DualObjectiveValue, T::Type)
+    value = zero(T) # sum will not work if there are zero constraints
+    for (F, S) in MOI.get(model, MOI.ListOfConstraints())
+        value += dual_objective_value(model, F, S, T, attr.result_index)::T
+    end
+    if MOI.get(model, MOI.ObjectiveSense()) != MOI.MAX_SENSE
+        value = -value
+    end
+    if !is_ray(MOI.get(model, MOI.DualStatus()))
+        # The objective constant should not be present in rays
+        F = MOI.get(model, MOI.ObjectiveFunctionType())
```

Comment:
Is this logic correct for quadratic objectives? Isn't there a contribution from a quadratic objective to the dual? https://en.wikipedia.org/wiki/Quadratic_programming#Lagrangian_duality

Reply:
There is no constant in the objective in the wiki article.
The constant in the primal objective is simply added as a constant in the dual objective